src/main/java/org/openlmis/stockmanagement/service/SourceDestinationBaseService.java

```diff
@@ -343,4 +356,35 @@ private List<ValidSourceDestinationDto> createAssignmentDto(
 
     return Pagination.getPage(validDestinations, pageable, foundPage.getTotalElements());
   }
+
+  private <T extends SourceDestinationAssignment> Page<ValidSourceDestinationDto>
+      findEnabledAssignments(SourceDestinationAssignmentRepository<T> repository,
+                         Profiler profiler, Pageable pageable) {
+    profiler.start("FIND_ENABLED_ASSIGNMENTS");
+
+    List<? extends SourceDestinationAssignment> assignments = repository.findAll();
```

Comment:
It would be more optimal to make special method in the repository to filter enabled assignments with fetching in one query. But from code perspective this solution looks cleaner.
I would leave this solution as it is but note this one could be optimized in the future if performance is not enough.
But if you feel that you want to optimize in this PR or you already know that this would be a big performance pitfall then please feel free to optimize it.

Reply:
I agree. I tried to replace this with a custom query to use database pagination but due to the generic implementation of this method it is not that easy. In case of a significant drop in performance, it will be necessary to refactor this method using e.g. javax.persistence.criteria.CriteriaQuery which, thanks to the flexibility in building queries, should work in this case.

Reply:
So do you want to leave it as it is or refactor to use Criteria Query?

Reply:
Ok.